### PR TITLE
Apply OWASP dependency-check to analyze jar file

### PR DESCRIPTION
**Pull request by bjk7119**
## Description

Apply OWASP dependency-check to analyze jar file

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Refactoring, Maintenance
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/fosslight_binary/_binary_dao.py [hunk 1]
```diff
@@ -19,6 +20,24 @@
 logger = logging.getLogger(constant.LOGGER_NAME)
 
 
+def move_owasp_info_to_comment_column(binary_item, row, bin_info_list):
```
Comment (soimkim) [inline src/fosslight_binary/_binary_dao.py:23]: Let's remove bin_info_list

Comment (bjk7119) [reply src/fosslight_binary/_binary_dao.py]: Okay, but I changed the code and removed the function.
Check and review plz.

File: src/fosslight_binary/_binary.py [hunk 2]
```diff
@@ -14,11 +15,29 @@ class OssItem:
...
+    def set_oss_comment(self, oss_comment, Binary_comment=""):
+        self.comment += oss_comment
+        if Binary_comment != "":
+            Binary_comment += '/ '
+        self.comment = f"{Binary_comment}{self.comment}"
+
+    def set_oss_exclude(self, value):
+        self.oss_exclude = _EXCLUDE_TRUE_VALUE if value else ""
+
+    def get_print_oss_comment(self):
```
Comment (soimkim) [inline src/fosslight_binary/_binary.py:39]: get_print_oss_comment -> get_oss_comment

File: src/fosslight_binary/_binary.py [hunk 3]
```diff
@@ -14,11 +15,29 @@ class OssItem:
...
+    comment = ""
+    oss_exclude = ""
 
-    def __init__(self, name, version, license):
+    def __init__(self, name, version, license, dl_url=""):
         self.name = name
         self.version = version
         self.license = license
+        self.dl_url = dl_url
+        self.oss_exclude = ""
```
Comment (soimkim) [inline src/fosslight_binary/_binary.py:27]: oss_exclude -> exclude

File: src/fosslight_binary/_binary.py (hunk 2 above)
Comment (bjk7119) [reply src/fosslight_binary/_binary.py]: changed

File: src/fosslight_binary/_binary.py [hunk 4]
```diff
@@ -29,6 +44,8 @@ class BinaryItem:
     checksum = _TLSH_CHECKSUM_NULL
     oss_items = []
     exclude = ""
+    binary_comment = ""
```
Comment (soimkim) [inline src/fosslight_binary/_binary.py:47]: binary_comment -> comment

File: src/fosslight_binary/_binary.py [hunk 5]
```diff
@@ -29,6 +44,8 @@ class BinaryItem:
...
+    found_in_db = False
 
     def __init__(self, value):
         self.exclude = ""
```
Comment (soimkim) [inline src/fosslight_binary/_binary.py:51]: self.exclude = "" -> self.exclude = False

File: src/fosslight_binary/_binary.py [hunk 6]
```diff
@@ -29,6 +44,8 @@ class BinaryItem:
     checksum = _TLSH_CHECKSUM_NULL
     oss_items = []
     exclude = ""
```
Comment (soimkim) [inline src/fosslight_binary/_binary.py:46]: exclude = "" -> exclude = False

**Review by soimkim — CHANGES_REQUESTED** [review]

File: src/fosslight_binary/binary_analysis.py [hunk 7]
```diff
@@ -103,13 +110,38 @@ def get_file_list(path_to_find):
...
+    not_found_bin = []
+
+    # key : file_path / value : oss_list for one binary
+    for key, value in owasp_items.items():
+        found = False
+        for bin in bin_list:
+            if bin.binary_strip_root == key:
+                for oss in value:
+                    bin.set_oss_items(oss, False)
+                found = True
```
Comment (soimkim) [inline src/fosslight_binary/binary_analysis.py:132]: AS-IS: 
```
for oss in value:
                    bin.set_oss_items(oss, False)
```
TO-BE:
```
bin.set_oss_items(value, False)
```

File: src/fosslight_binary/binary_analysis.py [hunk 8]
```diff
@@ -103,13 +110,38 @@ def get_file_list(path_to_find):
...
+                break
+
+        if not found:
+            bin_item = BinaryItem(os.path.abspath(key))
+            bin_item.binary_name_without_path = os.path.basename(key)
+            bin_item.binary_strip_root = key
+            for oss in value:
+                bin_item.set_oss_items(oss)
+            not_found_bin.append(bin_item)
```
Comment (soimkim) [inline src/fosslight_binary/binary_analysis.py:141]: AS-IS: 
```
for oss in value:
                    bin_item.set_oss_items(oss)
```
TO-BE:
```
bin_item.set_oss_items(value)
```

**Review by soimkim — CHANGES_REQUESTED** [review]

File: src/fosslight_binary/_binary.py [hunk 9]
```diff
@@ -42,30 +59,42 @@ def __init__(self, value):
...
 
-    def set_oss_items(self, value):
-        self.oss_items.append(value)
+    def set_oss_items(self, new_oss, exclude_old=False, exclude_msg=""):
+        if exclude_old:
+            for old_oss in self.oss_items:
+                old_oss.set_oss_exclude(True)
+                old_oss.set_oss_comment(exclude_msg)
+        # Append New input OSS
+        self.oss_items.append(new_oss)
```
Comment (soimkim) [inline src/fosslight_binary/_binary.py:68]: self.oss_items.extend(new_oss)

File: src/fosslight_binary/_jar_analysis.py [hunk 10]
```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020 LG Electronics Inc.
```
Comment (soimkim) [inline src/fosslight_binary/_jar_analysis.py:3]: Copyright (c) 2022 LG Electronics Inc.

File: src/fosslight_binary/binary_analysis.py [hunk 11]
```diff
@@ -127,9 +159,15 @@ def find_binaries(path_to_find_bin, output_dir, format, _include_file_command, d
                           result_log=_result_log,
                           exit=True)
 
-        total_file_cnt, file_list = get_file_list(path_to_find_bin)
+        total_file_cnt, file_list, found_jar = get_file_list(path_to_find_bin)
         total_bin_cnt, return_list = return_bin_only(file_list, _include_file_command)
 
+        # Run OWASP Dependency-check
+        if found_jar is True:
```
Comment (soimkim) [inline src/fosslight_binary/binary_analysis.py:166]: if found_jar is True: ->  if found_jar:

File: src/fosslight_binary/_jar_analysis.py [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+                                oss_name = product_info['value']
+                            if oss_ver == "" and (product_info['name'] == 'Implementation-Version' or product_info['name'] == 'Bundle-Version'):
+                                oss_ver = product_info['value']
+
+                oss = OssItem(oss_name, oss_ver, oss_license, oss_dl_url)
+                oss.set_oss_comment("OWASP Result. ")
+
+                owasp_item = owasp_items.get(file_with_path)
+                if owasp_item:
+                    owasp_item.append(oss)
```
Comment (soimkim) [inline src/fosslight_binary/_jar_analysis.py:118]: owasp_item.append(oss) -> owasp_items.append(oss)

File: src/fosslight_binary/_jar_analysis.py [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+        if not split_lic.startswith('http'):
+            license = split_lic.replace(',', '')
+        else:
+            license = license_raw
+
+    return license
+
+
+def ananlyze_jar_file(path_to_find_bin):
+    owasp_item = []
```
Comment (soimkim) [inline src/fosslight_binary/_jar_analysis.py:45]: remove owasp_item = []

File: src/fosslight_binary/binary_analysis.py [hunk 14]
```diff
@@ -127,9 +159,15 @@ def find_binaries(path_to_find_bin, output_dir, format, _include_file_command, d
...
+            logger.info("Run OWASP Dependency-check to analyze .jar file")
+            owasp_items = ananlyze_jar_file(path_to_find_bin)
+            return_list = merge_binary_list(owasp_items, return_list)
```
Comment (soimkim) [inline src/fosslight_binary/binary_analysis.py:169]: if owasp_items:
    return_list = merge_binary_list(owasp_items, return_list)

File: src/fosslight_binary/_binary.py [hunk 15]
```diff
@@ -42,33 +59,46 @@ def __init__(self, value):
...
             for oss in self.oss_items:
+                exclude = _EXCLUDE_TRUE_VALUE if (self.exclude or oss.exclude) else ""
                 print_rows.append([self.binary_strip_root, oss.name, oss.version,
-                                   oss.license, '', '', '', self.exclude, ''])
+                                   oss.license, oss.dl_url, '', '', exclude, oss.comment])
         else:
+            exclude = _EXCLUDE_TRUE_VALUE if self.exclude else ""
             print_rows.append([self.binary_strip_root, '',
-                               '', '', '', '', '', self.exclude, ''])
+                               '', '', '', '', '', exclude, ''])
```
Comment (soimkim) [inline src/fosslight_binary/_binary.py:101]: exclude -> self.exclude

**Review by soimkim — APPROVED** [review]

**Pull request merged**